catalyst-gateway-crates/c509-certificate/src/c509_oid.rs

```diff
@@ -0,0 +1,213 @@
+//! C509 OID provides an encoding and decoding of C509 Object Identifier (OID).
+//! Please refer to [RFC9090](https://datatracker.ietf.org/doc/rfc9090/) for OID encoding
+//! Please refer to [CDDL Wrapping](https://datatracker.ietf.org/doc/html/rfc8610#section-3.7)
+//! for unwrapped types.
+
+use anyhow::Result;
+use asn1_rs::oid;
+use minicbor::{data::Tag, decode, encode::Write, Decode, Decoder, Encode, Encoder};
+use oid_registry::Oid;
+
+use crate::tables::IntegerToOidTable;
+
+/// IANA Private Enterprise Number (PEN) OID prefix.
+const PEN_PREFIX: Oid<'static> = oid!(1.3.6 .1 .4 .1);
+
+/// Tag number representing IANA Private Enterprise Number (PEN) OID.
+const OID_PEN_TAG: u64 = 112;
+
+/// A strut of C509 OID with Registered Integer Encoding/Decoding.
+#[derive(Debug, Clone, PartialEq)]
+pub struct C509oidRegistered {
```

Comment:
```suggestion
pub struct C509OidRegistered {
```

Reply:
Capital `O` might be confused with `0` so `o` is used